### PR TITLE
enable lbaas based on new systemd_service module

### PR DESCRIPTION
**Pull request by panxia6679**

### CODE REVIEW

File: roles/neutron-common/meta/main.yml [hunk 1]
```diff
@@ -30,3 +30,10 @@ dependencies:
     python_post_dependencies: "{{ neutron.distro.python_post_dependencies }}"
     rootwrap: True
     when: openstack_install_method == 'distro'
+  - role: openstack-distro
+    project_name: neutron-lbaas
+    dependent_packages: "{{ neutron.osp_sys_packages }}"
+    rootwrap: True
+    when: os == 'rhel' and openstack_install_method == 'distro' and ((neutron.lbaas.enabled == "smart" and
```
Comment (nirajdp76) [inline roles/neutron-common/meta/main.yml]: might be more readable:
when:
  - os == 'rhel'
  - openstack_install_method == 'distro'
  - ((neutron.lbaas.enabled == "smart" and groups['controller'][0] not in groups['compute']) or
        neutron.lbaas.enabled|bool)

File: roles/neutron-data-network/tasks/main.yml [hunk 2]
```diff
@@ -167,22 +167,26 @@
     - haproxy.loadbalancer.j2
     - haproxy_proxies.j2
   notify: restart neutron lbaas agent
-  when: (neutron.lbaas.enabled == "smart" and
+  when: (os != 'rhel' or openstack_install_method != 'distro') and ((neutron.lbaas.enabled == "smart" and
```
Comment (nirajdp76) [inline roles/neutron-data-network/tasks/main.yml]: is this needed for openstack_install_method == 'source'  (i.e ubuntu and centos ?) or is needed for source ((ubuntu, centos) and package (ubuntu giftwrap)?

if is latter than might want to have only openstack_install_method != 'distro'  in 1st part of the when clause

File: roles/neutron-data-network/tasks/main.yml [hunk 3]
```diff
@@ -167,22 +167,26 @@
...
     group: neutron
-    mode: 0644
-  when: (neutron.lbaas.enabled == "smart" and
+    state: link
+    force: true
+  with_items:
+    - haproxy.loadbalancer.j2
+    - haproxy_proxies.j2
+  notify: restart neutron lbaas agent
+  when: os == 'rhel' and openstack_install_method == 'distro' and ((neutron.lbaas.enabled == "smart" and
```
Comment (nirajdp76) [inline roles/neutron-data-network/tasks/main.yml]: no need to add os =='rhel'

Comment (panxia6679) [inline roles/neutron-data-network/tasks/main.yml]: thanks. updated code.

File: roles/neutron-data-network/tasks/main.yml (hunk 2 above)
Comment (panxia6679) [inline roles/neutron-data-network/tasks/main.yml]: yep, checking distro would be enough.

File: roles/neutron-common/meta/main.yml (hunk 1 above)
Comment (panxia6679) [inline roles/neutron-common/meta/main.yml]: thanks for your comment. that is really more readable.

**Comment by nirajdp76** [comment]
ok to test

**Pull request merged**